src/diffusers/models/attention_processor.py

```diff
@@ -378,7 +378,7 @@ def set_processor(self, processor: "AttnProcessor", _remove_lora: bool = False)
             _remove_lora (`bool`, *optional*, defaults to `False`):
                 Set to `True` to remove LoRA layers from the model.
         """
-        if not USE_PEFT_BACKEND and hasattr(self, "processor") and _remove_lora and self.to_q.lora_layer is not None:
```

Comment:
Why remove `and self.to_q.lora_layer is not None:` here?

Reply:
ah because when `not USE_PEFT_BACKEND == True` the `self.to_q` is an `nn.Linear` so it has no `.lora_layer` I think this test is redundant

Reply:
Hmm I don't think that's correct. If we don't use PEFT then `self.to_q` can have a `lora_layer`

Reply:
I think this part needs to be reverted the rest looks good

Reply:
ah yes you are right! reverting 🙇🏽